mongodb/install.sls

```diff
@@ -0,0 +1,37 @@
+{% from "mongodb/map.jinja" import mongodb with context %}
+
+include:
+  - .service
+  - .configure
+
+
+{% if mongodb.install_pkgrepo %}
+add_mongodb_package_repository:
+  pkgrepo.managed:
+    - name: {{ mongodb.repo }}
+    - humanname: MongoDB Repository
+    {% if mongodb.get('repo') %}
+    - dist: {{ mongodb.repo_dist }}
+    {% endif %}
+    - keyid: {{ mongodb.key }}
+    - keyserver: {{ mongodb.keyserver }}
+    - refresh_db: True
+{% endif %}
+
+install_packages:
+  pkg.installed:
+    - pkgs: {{ mongodb.pkgs }}
+    - refresh: True
+    - install_recommends: True
+    - require:
+      - pkgrepo: add_mongodb_package_repository
+    - require_in:
+      - service: mongodb_service_running
+      - file: copy_mongodb_key_file
+      - service: mongodb_service_running
+
+{% for egg in mongodb.pip_pkgs %}
+install_{{ egg }}_pip_package:
+  pip.installed:
+    - name: {{ egg }}
+{% endfor %}
```

Comment:
For this you can just do:
```salt
install_pip_dependencies:
  pip.installed:
    - name: mongodb dependencies
    - pkgs: {{ mongodb.pip_pkgs }}
```

You don't need to use the for loop. This is an undocumented feature of this state (https://github.com/blarghmatey/salt/blob/develop/salt/states/pip_state.py#L536-L544)